src/DynamoCoreWpf/Views/FileTrust/FileTrustWarning.xaml.cs

```diff
@@ -47,6 +47,15 @@ public FileTrustWarning(DynamoView dynamoViewWindow)
             SetUpPopup();
 
             HomeWorkspaceModel.WorkspaceClosed += CloseWarningPopup;
+            dynViewModel.PropertyChanged += DynViewModel_PropertyChanged;
+        }
+
+        private void DynViewModel_PropertyChanged(object sender, System.ComponentModel.PropertyChangedEventArgs e)
+        {
+            if (e.PropertyName == "CurrentSpace" && !((sender as DynamoViewModel).HomeSpaceViewModel as HomeWorkspaceViewModel).IsCurrentSpace)
+            {
+                IsOpen = false;
+            }
```

Comment:
Do we need an else case to make it visible again when switching back?

Reply:
No because the HomeworkSpace do that here UpdateRunStatusMsgBasedOnStates , it looks like the trust ui was focused only with one Workspace.